src/lib/components/workflow/pending-activity/workflow-pending-activity.svelte

```diff
@@ -0,0 +1,220 @@
+    {/if}
+    {#if activity.heartbeatDetails}
+      {@render detail(translate('workflows.heartbeat-details'), heartbeat)}
+    {/if}
+  </div>
+</Card>
+
+{#snippet detail(label: string, value: string | number | Snippet)}
+  <div>
+    <p class="text-secondary">{label}</p>
+    {#if typeof value === 'string' || typeof value === 'number'}
+      {value}
+    {:else}
+      {@render value?.()}
+    {/if}
+  </div>
+{/snippet}
+
+{#snippet heartbeat()}
+  <PayloadDecoder
+    value={activity.heartbeatDetails}
+    let:decodedValue
+    key="payloads"
+  >
+    <CodeBlock
+      content={decodedValue}
+      copyIconTitle={translate('common.copy-icon-title')}
+      copySuccessIconTitle={translate('common.copy-success-icon-title')}
+    />
+  </PayloadDecoder>
+{/snippet}
+
+{#snippet failures()}
+  <Accordion
+    title={activity.lastFailure?.stackTrace
+      ? 'Last Failure with Stack Trace'
+      : 'Last Failure'}
+  >
+    <div class="flex flex-col gap-2">
+      <div class="flex flex-1 flex-col">
+        {#if activity.lastFailure}
+          <p class="text-subtle">{translate('workflows.last-failure')}</p>
+          <CodeBlock
+            class="pb-2"
+            content={stringifyWithBigInt(
+              omit(activity.lastFailure, 'stackTrace'),
+            )}
+            copyIconTitle={translate('common.copy-icon-title')}
+            copySuccessIconTitle={translate('common.copy-success-icon-title')}
+          />
+        {/if}
+      </div>
+      <div class="flex flex-1 flex-col">
+        {#if activity.lastFailure?.stackTrace}
+          <p class="text-subtle">{translate('common.stack-trace')}</p>
+          <CodeBlock
+            language="text"
+            content={activity.lastFailure.stackTrace}
+            copyIconTitle={translate('common.copy-icon-title')}
+            copySuccessIconTitle={translate('common.copy-success-icon-title')}
+          />
+        {/if}
+      </div>
+    </div>
+  </Accordion>
+{/snippet}
+
+{#snippet scheduleEventLink()}
+  <Link
+    class="text-primary hover:underline"
+    href={routeForEventHistoryEvent({
+      namespace: namespace,
+      eventId: scheduledEvent?.id,
+      workflow,
+      run,
+    })}>{scheduledEvent?.id}</Link
+  >
+{/snippet}
+
+{#snippet nextRetry()}
+  <div class="flex items-center gap-1">
+    {formatDate(activity.scheduledTime, $timeFormat, {
+      relative: $relativeTime,
+    })}
+    <strong
+      >{toTimeDifference({
+        date: activity.scheduledTime,
+        negativeDefault: translate('workflows.no-retry'),
+      })}</strong
+    >
+  </div>
+{/snippet}
+
+{#snippet attempts()}
+  <div class="flex items-center gap-1">
+    <Badge class="mr-1" type={failed ? 'danger' : 'default'}>
+      <Icon class="mr-1 {failed && 'font-bold text-red-400'}" name="retry" />
+      {activity.attempt} of {formatMaximumAttempts(activity.maximumAttempts)}
+    </Badge>
+    {#if activity.maximumAttempts}
+      <span class="text-sm text-subtle"
```

Comment:
`text-secondary` is a little bit easier to read here ⬇️ 

| Before | After |
|-|-|
|![Screenshot 2025-05-01 at 10 38 19 AM](https://github.com/user-attachments/assets/5e942a65-ab95-49d4-b5e3-9039202d662b)|![Screenshot 2025-05-01 at 10 38 09 AM](https://github.com/user-attachments/assets/100933a9-6950-41c0-bf91-7be9b0420bcf)|

```suggestion
      <span class="text-sm text-secondary"
```